pkg/stanza/fileconsumer/reader.go

```diff
@@ -44,6 +46,13 @@ type Reader struct {
 	file           *os.File
 	fileAttributes *FileAttributes
 	eof            bool
+
+	HeaderFinalized  bool
+	HeaderAttributes map[string]any
```

Comment:
Hmm, trying to think how we can do that.

It needs to be an exported field on the reader, so that it's serialized for storage. We could try to export the fileAttributes, I guess, and serialize that? I'll try it.

Reply:
Tried combining, I like that it dedupes the two maps, so we don't have to think too much about keeping them in sync.